src/main/scala/com/fulcrumgenomics/bam/SamRecordClipper.scala

```diff
@@ -289,6 +290,96 @@ class SamRecordClipper(val mode: ClippingMode, val autoClipAttributes: Boolean)
     }
   }
 
+  /** Clips overlapping read pairs, where both ends of the read pair are mapped to the same chromosome, and in FR orientation.
+    *
+    * @param rec the read
+    * @param mate the mate
+    * @return the number of overlapping bases to that were clipped on the record and mate respectively (3' end in sequencing order)
+    */
+  def clipOverlappingReads(rec: SamRecord, mate: SamRecord): (Int, Int) = {
+    if (rec.matesOverlap.contains(false)) (0, 0) // do not overlap, don't clip
+    else {
+      require(rec.isFrPair, "Only know how to clip FR overlapping read pairs")
+      if (rec.negativeStrand) clipOverlappingReads(rec=mate, mate=rec).swap // don't for get to swap the results since we swapped inputs
+      else {
+        var numOverlappingBasesReadOne: Int = 0
+        var numOverlappingBasesReadTwo: Int = 0
+        // What we really want is to trim by the number of _reference_ bases not read bases,
+        // in order to eliminate overlap.  We could do something very complicated here, or
+        // we could just trim read bases in a loop until the overlap is eliminated!
+        while (rec.end >= mate.start && rec.mapped && mate.mapped) {
+          val lengthToClip = rec.end - mate.start + 1
+          val firstHalf    = lengthToClip / 2
+          val secondHalf   = lengthToClip - firstHalf // safe guard against rounding on odd lengths
+          numOverlappingBasesReadOne += this.clip3PrimeEndOfAlignment(rec, firstHalf)
+          numOverlappingBasesReadTwo += this.clip3PrimeEndOfAlignment(mate, secondHalf)
+        }
+        (numOverlappingBasesReadOne, numOverlappingBasesReadTwo)
+      }
+    }
+  }
+
+  /** Clips a read that sequences past the start (in sequencing order) of the mate, and in FR orientation.
+    *
+    * @param rec the read
+    * @param mate the mate
+    * @param doNotClip do not clip the reads, only return the number of bases that would be clipped
```

Comment:
I could be convinced to update all the methods to have a mode where they don't actually clip, but just return how many additional bases _they would clip_.